Programs/brlapi_client.c

```diff
@@ -1658,20 +1659,46 @@ static int getControllingTty(void)
 #endif /* WINDOWS */
 
 #ifdef linux
-  {
-    FILE *stream;
+  do {
+    int fd = open("/proc/self/stat", O_RDONLY, 0);
+    if (fd < 0) {
+      break;
+    }
 
-    if ((stream = fopen("/proc/self/stat", "r"))) {
-      int vt;
-      int ok = fscanf(stream, "%*d %*s %*c %*d %*d %*d %d", &tty) == 1;
-      fclose(stream);
+    char buf[256];
+    ssize_t read_bytes = read(fd, buf, sizeof(buf) - 1);
+    close(fd);
+    if (read_bytes < 0) {
+      break;
+    }
+    /* Append null at the end of buffer to ensure that read doesn't
+     * overflows.*/
+    buf[read_bytes] = '\0';
+
+    /* The line format is "$pid ($comm) $state ...
+     * where comm may contain space and `)` as part of its name, ex:
+     * 12345 (some proc (foo) S 123 456) R ...
```

Comment:
you can drop ` S 123 456`, the ` (foo)` part alone is enough

Reply:
Done